azurerm/internal/services/synapse/validate/synapse_workspace.go

```diff
@@ -0,0 +1,51 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+func SynapseWorkspaceName(i interface{}, k string) (warnings []string, errors []error) {
```

Comment:
Maybe we should just return a single error instead of slice of errors? Instead of appending the errors maybe expose a boolean like `hasErrors` and then at the end of the function test to see if the bool is true or not, if `true` return the single error string? The reason I am suggesting this is because the first and second error messages have duplicate information about the `ondemand` suffix.

Reply:
I have resolved it by removing the duplicate information in the first error message. Because the regex could not detect it